### PR TITLE
Add initial GitHub actions based CI recipe.

### PR DESCRIPTION
**Pull request by s-ludwig**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        name: Test Suite
+        strategy:
+            matrix:
+                os: [ubuntu-latest, windows-latest, macOS-latest]
+                dc: [dmd-latest, ldc-latest]
+                arch: [x86_64]
+                include:
+                  - { config:select }
+                  - { config:epoll }
+                  - { config:cfrunloop }
```
Comment (Geod24) [inline .github/workflows/ci.yml:24]: Github CI, unlike Travis, is does not allow you to add rows to the matrix. So this probably won't work.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    branches:
+      - master
+  push:
+    branches:
+      - master
+      - github_actions
+
+jobs:
+    test:
+        name: Test Suite
```
Comment (Geod24) [inline .github/workflows/ci.yml:15]: I try to use short names because they get printed and then truncated...

File: .github/workflows/ci.yml (hunk 1 above)
Comment (Geod24) [reply .github/workflows/ci.yml:24]: You want:
```yaml
- { os: whatever, config: name }
```

For all three OSes

Comment (s-ludwig) [reply .github/workflows/ci.yml]: I've gone the other way around now and used `config: [select, epoll, cfrunloop]`, with an `exclude:` to remove bad configurations, which seems to work right. Just Windows doesn't produce any test output.

**Comment by s-ludwig** [comment]
Okay, seems to work now! I've deleted the associated AppVeyor project, too, although it is still being displayed here.

**Comment by Geod24** [comment]
Nice! The only thing worrying me about GH actions is that you can't re-trigger a single matrix row, and since we sometimes have flakey CI, it can be quite painful.
The other thing is, we need to adapt Buildkite I think.

**Pull request merged**